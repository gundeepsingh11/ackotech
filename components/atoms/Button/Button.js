/* eslint-disable react/default-props-match-prop-types */
// @flow
/**
 * Button component to present primary button and handle all the click events
 * @module Button
 */
import type from 'react';
import Link from '../Anchor';
import withStyles from '../../../lib/withStyles';
import styles from './Button.style';

const Button = ({
  className,
  children,
  inheritedStyles,
  type,
  primary,
  secondary,
  elementRef,
  disabled,
  ariaLabel,
  value,
  buttonWithLink,
  url,
  ...others
}) =>
  /* eslint-disable react/button-has-type */
  buttonWithLink ? (
    <Link to={url}>
      <button
        className={className}
        type={type}
        {...others}
        ref={elementRef}
        disabled={disabled}
        aria-label={ariaLabel || null}
        value={value}>
        {children}
      </button>
    </Link>
  ) : (
    <button
      className={className}
      type={type}
      {...others}
      ref={elementRef}
      disabled={disabled}
      aria-label={ariaLabel || null}
      value={value}>
      {children}
    </button>
  );
/* eslint-enable react/button-has-type */

Button.defaultProps = {
  inheritedStyles: '',
  type: 'button',
  primary: false,
  secondary: false,
  elementRef: undefined,
  disabled: false,
  ariaLabel: '',
};

export default withStyles(Button, styles);

export { Button as ButtonVanilla };
