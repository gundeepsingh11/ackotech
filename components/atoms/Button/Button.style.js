import { css } from 'styled-components';

const ButtonStyles = css`
  display: inline-block;
  position: relative;
  text-align: center;
  text-decoration: none;
  outline: 0;
  z-index: 1;
  color: ${props => props.theme.textColorGrey};
  background: transparent;
  border: none;
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  opacity: ${props => (props.disabled ? '0.3' : '1')};
  transition: 0.5s;

  &::after {
    content: '';
    width: 15px;
    height: 7px;
    display: inline-block;
    background: url('static/img/icon-page-link.svg') no-repeat;
    background-size: 100%;
    
    transition: right 0.2s;
    position: relative;
    top: -1px;
    right: 0;
    margin-left: 15px;
  }

  &:hover {
    &::after {
      /* background: ${props => props.theme.buttonHoverColor}; */
      right: -5px;
    }
  }

  &:focus {
    outline: 1px dashed ${props => props.theme.colorOutline};
  }

  &:focus:not(.focus-visible) {
    outline: none;
  }


  ${props =>
    // eslint-disable-next-line operator-linebreak
    props.secondary &&
    css`
      color: ${props.theme.textColorPrimary};

      &::before {
        background: ${props.theme.textColorWhite};
        border: 2px solid ${props.theme.textColorPrimary};
      }

      &:hover {
        color: ${props.theme.textColorWhite};

        &::before {
          background: ${props.theme.textColorPrimary};
        }
      }

      &::after {
        content: '';
        border: none;
      }
    `}
`;
export default ButtonStyles;
