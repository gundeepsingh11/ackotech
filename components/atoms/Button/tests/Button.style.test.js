import { shallow } from 'enzyme';
import React from 'react';
import 'jest-styled-components';

import Button from '../Button';

const props = {
  secondary: 'true',
};

describe('Button', () => {
  let button = '';

  test('should render secondary correctly', () => {
    button = shallow(<Button {...props} />);
    expect(button).toMatchSnapshot();
  });

  test('should render correctly ', () => {
    button = shallow(<Button />);
    expect(button).toMatchSnapshot();
  });

  test('should render disabled', () => {
    button = shallow(<Button disabled>Submit</Button>);
    expect(button.find({ disabled: true })).toHaveLength(1);
  });
});
