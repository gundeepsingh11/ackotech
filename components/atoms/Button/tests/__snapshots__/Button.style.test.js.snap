// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly  1`] = `
.c0 {
  display: inline-block;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  z-index: 1;
  padding: 0.75rem 1.5rem;
  min-width: 240px;
  background: transparent;
  border: none;
  cursor: pointer;
  opacity: 1;
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.c0::before {
  content: "";
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  -webkit-transform: skew(-21deg);
  -ms-transform: skew(-21deg);
  transform: skew(-21deg);
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.c0::after {
  content: "";
  position: absolute;
  left: 6px;
  top: 6px;
  width: 100%;
  height: 100%;
  background: transparent;
  border: 2px solid;
  z-index: -2;
  -webkit-transform: skew(-21deg);
  -ms-transform: skew(-21deg);
  transform: skew(-21deg);
}

.c0:focus {
  outline: 1px dashed;
}

.c0:focus:not(.focus-visible) {
  outline: none;
}

<Button
  ariaLabel=""
  className="c0"
  disabled={false}
  inheritedStyles=""
  primary={false}
  secondary={false}
  type="button"
/>
`;

exports[`Button should render secondary correctly 1`] = `
.c0 {
  display: inline-block;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  z-index: 1;
  padding: 0.75rem 1.5rem;
  min-width: 240px;
  background: transparent;
  border: none;
  cursor: pointer;
  opacity: 1;
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.c0::before {
  content: "";
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  -webkit-transform: skew(-21deg);
  -ms-transform: skew(-21deg);
  transform: skew(-21deg);
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.c0::after {
  content: "";
  position: absolute;
  left: 6px;
  top: 6px;
  width: 100%;
  height: 100%;
  background: transparent;
  border: 2px solid;
  z-index: -2;
  -webkit-transform: skew(-21deg);
  -ms-transform: skew(-21deg);
  transform: skew(-21deg);
}

.c0:focus {
  outline: 1px dashed;
}

.c0:focus:not(.focus-visible) {
  outline: none;
}

.c0::before {
  border: 2px solid;
}

.c0::after {
  content: '';
  border: none;
}

<Button
  ariaLabel=""
  className="c0"
  disabled={false}
  inheritedStyles=""
  primary={false}
  secondary="true"
  type="button"
/>
`;
